src/app/qgswelcomepage.cpp

```diff
@@ -277,9 +340,86 @@ void QgsWelcomePage::showContextMenuForTemplates( QPoint point )
   menu->popup( mTemplateProjectsListView->mapToGlobal( point ) );
 }
 
+void QgsWelcomePage::showContextMenuForNews( QPoint point )
+{
+  QModelIndex index = mNewsFeedListView->indexAt( point );
+  if ( !index.isValid() )
+    return;
+
+  const int key = index.data( QgsNewsFeedModel::Key ).toInt();
+
+  QMenu *menu = new QMenu();
+
+  QAction *dismissAction = new QAction( tr( "Dismiss" ), menu );
+  connect( dismissAction, &QAction::triggered, this, [this, key]
+  {
+    mNewsFeedParser->dismissEntry( key );
+  } );
+  menu->addAction( dismissAction );
+  QAction *dismissAllAction = new QAction( tr( "Dismiss All" ), menu );
+  connect( dismissAllAction, &QAction::triggered, this, [this]
+  {
+    mNewsFeedParser->dismissAll();
+  } );
+  menu->addAction( dismissAllAction );
+  menu->addSeparator();
+  QAction *hideAction = new QAction( tr( "Hide QGIS News…" ), menu );
+  connect( hideAction, &QAction::triggered, this, [this]
+  {
+    if ( QMessageBox::question( this,  tr( "QGIS News" ), tr( "Are you sure you want to permanently hide QGIS news? This action can not be undone." ) ) == QMessageBox::Yes )
```

Comment:
TODO: make this undo-able, in a different PR that'd cover hiding/showing individual welcome page sections in the options dialog. 

Not part of this PR, but might be worth adding the TODO note to remember to change the string here.

Reply:
Or maybe like in QT creator "restore default layout" for the whole welcome page.

Reply:
I've added this setting to the QGIS options dialog, and updated the string accordingly